nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/prometheusutil/PrometheusMetricsUtil.java

```diff
@@ -84,7 +85,8 @@ public static CollectorRegistry createNifiMetrics(NiFiMetricsRegistry nifiMetric
         addProcessingPerformanceMetrics(nifiMetricsRegistry, status.getProcessingPerformanceStatus(),
                 instanceId, componentType, componentName, componentId, parentPGId);
 
-        nifiMetricsRegistry.setDataPoint(status.getProcessingNanos(), "TOTAL_TASK_DURATION",
+        // prometheus metric expects milliseconds
+        nifiMetricsRegistry.setDataPoint(status.getProcessingNanos() / NANOS_PER_MILLI, "TOTAL_TASK_DURATION",
```

Comment:
I decided to use double division rather than the TimeUnit conversions because prometheus ultimately uses a double to represent the metric value and we can tolerate floating point errors.